design-system/src/theme-provider.tsx

```diff
@@ -88,21 +88,7 @@ const useTheme = (parentSelector = defaultParentSelector) => {
     setTheme(parentSelectorRef.current()?.dataset.theme || 'default');
   }, []);
 
-  // So consumers don't have to provide 'parentSelector' again as
-  // they already provided it in the hook call
-  const updateTheme = useRef(
-    ({ newTheme, themeOverrides }: Omit<TApplyTheme, 'parentSelector'>) => {
-      applyTheme({
-        newTheme,
-        parentSelector: parentSelectorRef.current,
-        themeOverrides,
-      });
-      setTheme(newTheme || 'default');
-    }
-  );
-  return useMemo(() => {
-    return { theme, applyTheme: updateTheme.current };
-  }, [theme]);
+  return useMemo(() => theme, [theme]);
```

Comment:
It seems that `useMemo` is not useful anymore now.

```suggestion
  return theme;
```

Reply:
Thanks 👍 

It's been updated in this commit: e57b8f14cba6f13161def8d9b9801c42c122530e